### ISSUE TITLE
XPM edit button for includes is not working when site is not in root url 

### ISSUE THREAD

**Issue by jhorsman** [sdl/dxa-web-application-java#29]
The XPM edit button for include pages like the header is not working when the website root is not "/".

![button](https://cloud.githubusercontent.com/assets/1956935/13227320/57fad4b8-d996-11e5-8666-c502d4782203.png)

For example, when the website root is /en/ as in http://example.com/en/ , the webiste will render fine. When opening XPM and clicking the include edit button the website will show an "Page Not Found" error.
For this example, the webapp...
- runs in the Tomcat ROOT webapp
- has publication url "/en/" (in the publication properties)
- has Path "/en" in the cd_dynamic_conf.xml

I think the issue is related to the XpmButtonTag in 
https://github.com/sdl/dxa-web-application-java/blob/3f2852181614b5e03b2100b61b9dfb945822e627/dxa-framework/dxa-common-impl/src/main/java/com/sdl/webapp/common/impl/taglib/xpm/XpmButtonTag.java

```
String path = this.pageContext.getServletContext().getContextPath();
```
- The code says the root of the website is made up by the servlet context and context path.
  - the contextPath should be /en/ since the publication and cd_dynamic_conf are set to serve /en/. Or alternatively the /en/ should be added to that.
- This is the only part of the code using the `javax.servlet.jsp.PageContext`, other parts of DXA use the servletContext and contextPath in `com.sdl.webapp.common.api.WebRequestContext`

Tested with DXA 1.2.1

**Comment by jhorsman** [comment]
I do not see an easy fix for this. It is tempting to implement com.sdl.webapp.common.api.WebRequestContext instead of javax.servlet.jsp.PageContext, but that does not resolve the situation since the servletContext and contextPath of the WebRequestContext are still empty.

**Comment by jhorsman** [comment]
Maybe the solution is not so set the contextPath after all. I think the problem is with this line

     String editUrl = '/' + path + this.region.getXpmMetadata().get(RegionModelImpl.INCLUDED_FROM_PAGE_FILE_NAME_XPM_METADATA_KEY);

Here we get the "IncludedFromPageFileName" property from the region's XPM metadata, while we should get the "IncludedFromPageID" and then resolve the pageLink.

How do can we resolve a page TCM URI to a page URL with Tridion dynamic linking?

**Comment by jhorsman** [comment]
The issue is fixed in this branch. https://github.com/jhorsman/dxa-web-application-java/tree/feature/xpm-button-fix

I did not do a pull request since this fix is done and tested on DXA 1.2. You can pull it from here into development, or I will come back later to merge it to DXA 1.3 when I get my hands on a DXA 1.3 box.

**Comment by jhorsman** [comment]
The fix is now tested in DXA 1.2 on Tridion 2013 SP1. Once with a website running in http://localhost/ and one running in http://localhost/en/

**Comment by rpannekoek** [comment]
TSI-1855 (internal issue ID for tracking purposes)

**Comment by rpannekoek** [comment]
Fixed in DXA 1.6.
